plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol

```diff
@@ -41,10 +41,11 @@ contract PaymentExitGame is IExitProcessor, OnlyFromAddress, PaymentStandardExit
      * @notice Callback processes exit function for the PlasmaFramework to call
      * @param exitId The exit ID
      * @param token Token (ERC20 address or address(0) for ETH) of the exiting output
+     * @param processor The processExit initiator
      */
-    function processExit(uint168 exitId, uint256, address token) external onlyFrom(address(paymentExitGameArgs.framework)) {
+    function processExit(uint168 exitId, uint256, address token, address payable processor) external onlyFrom(address(paymentExitGameArgs.framework)) {
```

Comment:
is there a reason passing in this instead of using `msg.sender` to prevent auto frontrunning bot?

Also, do we want to check the processor is really the msg.sender or are we fine to be different people? If so, can we rename it to sth else, probably `exitProcessCompoentionTarget` (bear my bad English, I believe there is better wording)

Reply:
Oh so the processor here is taken to award him the bounty. Since there is an internal call from framework contract to the exit game contract, passing in the address of the caller and awarding the bounty. Thinking about doing the anti-front running at the end in a different task

Reply:
okay. now I rethink it might be better to explicit pass in args vs using `msg.sender` as a clearer dependency declaration and decoupling. So ignore the part why not `msg.sender`.